components/scream/scripts/test_all_scream.py

```diff
@@ -50,6 +50,20 @@ def __init__(self, cxx_compiler, f90_compiler, c_compiler, submit=False, paralle
         #  Sanity checks and helper structs setup  #
         ############################################
 
+        # Probe machine if none was specified
+        if self._machine is None:
+            # We could potentially integrate more with CIME here to do actual
+            # nodename probing.
+            if "CIME_MACHINE" in os.environ and is_machine_supported(os.environ["CIME_MACHINE"]):
+                self._machine = os.environ["CIME_MACHINE"]
+            else:
+                expect(self._machine is not None, "Machine is now required by test-all-scream")
```

Comment:
Nit-picky here, but the condition is guaranteed to fail (given the enclosing if), so might as well put `expect(False,"...")`. Or even better use `raise`?

Reply:
Done.